### PR TITLE
Implemented navigation drawer, changed screen titles

### PR DESCRIPTION
**Pull request by vshtuka**

### CODE REVIEW

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt [hunk 1]
```diff
@@ -25,16 +25,35 @@ class MosaicRepository(application: Application) {
     fun getMosaicsList(): LiveData<List<Drawable>> {
         return mosaicsList
     }
+    private val instagramUserName = MutableLiveData<String>()
+    fun getUserName(): LiveData<String> {
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: Please add access modifiers for each function and avoid package-visible modifiers

Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: Also please reformat according to code style

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -2,13 +2,13 @@
     <string name="app_name">SuperImagesViewer</string>
     <string name="facebook_app_id">140609281215175</string>
     <string name="fb_login_protocol_scheme">fb140609281215175</string>
-    <string name="facebook_login">Auth</string>
-    <string name="mosaic_view">Photos</string>
+<!--    <string name="facebook_login">Auth</string>-->
```
Comment (FAQEnD) [inline app/src/main/res/values/strings.xml]: It is better to delete strings

**Review by FAQEnD — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt (hunk 1 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt:29]: done

Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: done

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (vshtuka) [reply app/src/main/res/values/strings.xml]: done

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt (hunk 1 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt [hunk 3]
```diff
@@ -42,16 +43,20 @@ class MosaicRepository(application: Application) {
...
             null,
             HttpMethod.GET
         ) { findUserNameResponse ->
             val userNameResponse: JSONObject = findUserNameResponse.jsonObject
-            val userName = userNameResponse.getString("username")
-            instagramUserName.postValue(userName)
-        }.executeAsync()
+            instagramUserName = userNameResponse.getString("username")
+        }
+        val thread = Thread {
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: no no no, please use kotlin coroutines instead

File: app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt [hunk 4]
```diff
@@ -4,17 +4,27 @@ import android.app.Application
...
     val mosaicsList: LiveData<List<Drawable>>
     private val mosaicRepository: MosaicRepository = MosaicRepository(application)
 
     init {
         mosaicsList = mosaicRepository.getMosaicsList()
-        userName = mosaicRepository.getUserName()
+        refreshReferences(application)
+    }
+
+    private fun refreshReferences(application: Application) {
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt]: Please rename to something more specific

File: app/src/main/java/com/example/superimagesviewer/ui/MosaicFragment.kt [hunk 5]
```diff
@@ -41,10 +42,8 @@ class MosaicFragment : Fragment() {
     }
 
     private fun setFragmentTitle() {
-        val userNameObserver = Observer {instagramUserName: String ->
-            (requireActivity() as AppCompatActivity).supportActionBar?.title = instagramUserName
-        }
-        mosaicViewModel.userName.observe(viewLifecycleOwner, userNameObserver)
+        (requireActivity() as AppCompatActivity).supportActionBar?.title =
+            Settings.getUserName(requireContext())
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/ui/MosaicFragment.kt]: Why you removed Observer pattern?

**Review by FAQEnD — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt (hunk 3 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt (hunk 4 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt]: done

File: app/src/main/java/com/example/superimagesviewer/ui/MosaicFragment.kt (hunk 5 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/ui/MosaicFragment.kt]: returned Observer pattern back

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt [hunk 6]
```diff
@@ -18,23 +19,52 @@ import org.json.JSONObject
...
+        val token = AccessToken.getCurrentAccessToken()
+        GraphRequest(
+            token,
+            "/$INSTAGRAM_ID?fields=username",
+            null,
+            HttpMethod.GET
+        ) { findUserNameResponse ->
+            val userNameResponse: JSONObject = findUserNameResponse.jsonObject
+            if (userName != userNameResponse.getString("username")) {
+                instagramUserName.postValue(userNameResponse.getString("username"))
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: Please move serNameResponse.getString("username") to variable

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt [hunk 7]
```diff
@@ -18,23 +19,52 @@ import org.json.JSONObject
 
 class MosaicRepository(application: Application) {
 
-    val TAG = MosaicRepository::class.simpleName
+    private val TAG = MosaicRepository::class.simpleName
 
-    private val drawables = mutableListOf<Drawable>()
+    private val drawables = mutableListOf<InstagramImage>()
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: Please rename to something like "instagramImageList"

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt [hunk 8]
```diff
@@ -79,8 +110,9 @@ class MosaicRepository(application: Application) {
                     .build()
                 val drawable = imageLoader.execute(coilRequest).drawable
                 if (drawable != null) {
-                    drawables.add(drawable)
-                    mosaicsList.postValue(drawables)
+                    drawables.add(InstagramImage(drawable, imageUploadData))
+                    drawables.sortByDescending { it.uploadDataTime }
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: This is kind of Business logic, it should be passed at ViewModel level

**Review by FAQEnD — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/superimagesviewer/ui/ImageUploadDialogFragment.kt [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.fragment.app.DialogFragment
+
+class ImageUploadDialogFragment(private val message: String) : DialogFragment() {
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        return activity?.let {
+            val builder = AlertDialog.Builder(it)
+            builder.setMessage(message)
+                .setPositiveButton("OK") { dialog, id ->
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/ui/ImageUploadDialogFragment.kt]: Please move string "OK" to strings file

File: app/src/main/java/com/example/superimagesviewer/ui/InstagramPostFragment.kt [hunk 10]
```diff
@@ -1,60 +1,66 @@
...
-            view.findNavController().navigate(
-                InstagramPostFragmentDirections.actionInstagramPostFragmentToMosaicFragment()
-            )
-        }
-
-        uploadButton.setOnClickListener {
-            val photoUrl = editTextPhotoUrl.text.toString()
+        binding.uploadButton.setOnClickListener {
+            val photoUrl = binding.editTextPhotoUrl.text.toString()
             if (URLUtil.isValidUrl(photoUrl)) {
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/ui/InstagramPostFragment.kt]: Please swap if else statements like this:
if (!URLUtil.isValidUrl(photoUrl)) {
showDialog("Invalid or empty url")
return;
}
and here all your code if url is valid.

This required to simplify code readability and remove unnecessary braces

File: app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt [hunk 11]
```diff
@@ -1,16 +1,46 @@
...
+    public val drawablesList: LiveData<List<Drawable>>
+    public val userName: LiveData<String>
+    private val mosaicRepository: MosaicRepository = MosaicRepository()
 
     init {
-        mosaicsList = mosaicRepository.getMosaicsList()
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                if (mosaicRepository.isNetworkConnected(application)) {
+                    Log.d(TAG, "Successful internet connection")
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt]: I think this log is not required anymore

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt [hunk 12]
```diff
@@ -5,37 +5,53 @@ import android.content.Context
...
+        val token = AccessToken.getCurrentAccessToken()
+        GraphRequest(
+            token,
+            "/$INSTAGRAM_ID?fields=username",
+            null,
+            HttpMethod.GET
+        ) { findUserNameResponse ->
+            val userNameResponse: JSONObject = findUserNameResponse.jsonObject
+            val newUserName = userNameResponse.getString("username")
+            if (userName != newUserName) {
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: This logic should be in view model

**Review by FAQEnD — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt (hunk 6 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt (hunk 7 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt (hunk 8 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/ui/ImageUploadDialogFragment.kt (hunk 9 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/ui/ImageUploadDialogFragment.kt]: Done

File: app/src/main/java/com/example/superimagesviewer/ui/InstagramPostFragment.kt (hunk 10 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/ui/InstagramPostFragment.kt]: Done

File: app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt (hunk 11 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt]: Done

File: app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt (hunk 12 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt]: Done

**Review by FAQEnD — APPROVED** [review]

**Pull request merged**